pom.xml

```diff
@@ -134,6 +134,31 @@
             </resource>
         </resources>
 
+        <pluginManagement>
```

Comment:
Ok, Maybe I started with the wrong foot here, lets start over.

1. Today in trunk if you wanna run just checkstyle, you do "mvn checktyle:check", but since maven-checkstyle-plugin is under "reporting" group, the previous check wont read checkstyle.xml and will return 8421 violations.

2. Previous travis-ci config was running "mvn test" as the default ( https://docs.travis-ci.com/user/languages/java ) so my intention is to keep integration test as light as possible and avoid "trash" generated by "mvn site" since its not useful in pull request integration tests.

3. The way I found to achieve this is to configure the goal on build->plugins->maven-checkstyle-plugin and then run "mvn checktyle:check" with the parameter "failsOnError" activated ( so the build will fail ).

4. This works on console, but eclipse m2e plugin starting complaining about project and stops solving artifacts. The workaround is to move config to build->pluginManagement->plugins->maven-checkstyle-plugin.

So looks to me that we have two options:
1. Move checkstyle config to "reporting" group and do a mvn site on travis-ci.
2. Go with isolated goal as the pull request submitted.

That's your call.
Regards